pkg/api/types.go

```diff
@@ -2154,3 +2154,32 @@ type RangeAllocation struct {
 	// a single allocated address (the fifth bit on CIDR 10.0.0.0/8 is 10.0.0.4).
 	Data []byte `json:"data"`
 }
+
+// LockSpec is the specification of a resource lock.
+type LockSpec struct {
+	// Who currently has the lock.
+	HeldBy string `json:"heldby"`
+	// How long the lock is valid.
+	LeaseTime uint64 `json:"duration"`
+	// When the lock was first acquired.
+	AcquiredTime string `json:"atime,omitempty"`
+	// When the lock was last renewed.
+	RenewTime string `json:"rtime,omitempty"`
+}
+
+// Lock represents the configuration of a resource lock.
+type Lock struct {
```

Comment:
I still think Lock is better also, but mike made a good point ... If we can't agree... then one possible comprimise is to have the lock remove the "RenewTime" field, and have 

```
type LockSpec { HeldBy, AcquiredTime}
type LeaseSpec { HeldBy, AcquiredTime, LockSpec}
```
That should solve everyone's problem and will make for clean decoupling of semantics.